### PR TITLE
Upgrade to rails 7

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Review by mjgiarlo — APPROVED** [review]

File: app/views/layouts/application.html.erb
Comment (mjgiarlo) [inline app/views/layouts/application.html.erb:8]: This is a different version of bootstrap than is pinned in the importmap. Is that OK?

Comment (jcoyne) [reply app/views/layouts/application.html.erb:8]: Yeah, it works fine.

File: config/importmap.rb
Comment (mjgiarlo) [inline config/importmap.rb:3]: When/how often should we be doing this? I guess I'm fuzzy on the new importmap + sprockets combo, mostly in terms of workflow but also in terms of conventions (see prior question about discrepancy between bootstrap js and css). Not blocking the PR, just hoping to understand more before we merge this.

File: package.json
Comment (mjgiarlo) [inline package.json:2]: Do we still need package.json now that most of its contents have been removed?

File: app/views/layouts/application.html.erb
Comment (mjgiarlo) [reply app/views/layouts/application.html.erb:8]: Is there any reason to want diff versions of bootstrap? I.e., any reason not to bump the js pin to 4.6.1?

File: package.json
Comment (jcoyne) [reply package.json:2]: I don't think so, but it's possible someone will need something in the future. How about we follow up after to pursue this.

File: yarn.lock
Comment (jcoyne) [reply yarn.lock:2]: I don't think so, but let's pursue this separately.

File: config/importmap.rb
Comment (jcoyne) [reply config/importmap.rb:3]: you do this when you add a JS dependency. This is the alternative to bunding the js using node. It transfers this work to the client, which makes the app simpler to deploy.

File: app/views/layouts/application.html.erb
Comment (jcoyne) [reply app/views/layouts/application.html.erb:8]: As a follow up I was going to upgrade to 5.0. But no, there's no reason to have different versions.

Comment (mjgiarlo) [reply app/views/layouts/application.html.erb:8]: OK, no bigs.

**Pull request merged**
